bots/s3_only_allow_ssl.py

```diff
@@ -0,0 +1,178 @@
+"""
+## s3_only_allow_ssl
+What it does: Ensure that S3 Buckets enforce encryption of data transfers using Secure Sockets Layer (SSL)
+Usage: AUTO: s3_only_allow_ssl
+Note: The bot looks at the bucket policy and adds to the current policy the missing actions(s3:GetObject and s3:PutObject)
+      and the SSL statement.
+      if no policy in the bucket an SSL policy will add to the bucket
+Limitations: none
+"""
+
+import json
+from botocore.exceptions import ClientError
+
+
+def run_action(boto_session, rule, entity, params):
+    # create a s3 sessions
+    s3_client = boto_session.client('s3')
+
+    bucket_name = entity['name']
+    account_number = entity["accountNumber"]
+    policy_bucket = entity['policy']
+
+    try:
+        if policy_bucket == "null":  # s3 does not have a bucket policy
+            policy = json.dumps(make_new_policy(bucket_name, account_number))
+
+        else:  # bucket has a policy
+            res = get_correct_policy(policy_bucket)
+            first_Effect, second_Effect = get_actions(bucket_name, account_number, res)
+            if second_Effect is None:  # if there is only one Effect to add to the bucket policy
+                policy_bucket['Statement'].append(first_Effect)
+                policy = json.dumps(policy_bucket)
+            else:
+                policy_bucket['Statement'].append(first_Effect)
+                policy_bucket['Statement'].append(second_Effect)
+                policy = json.dumps(policy_bucket)
+        print(policy)
+        result = s3_client.put_bucket_policy(
+            Bucket=bucket_name,
+            Policy=policy
+        )
+
+        responseCode = result['ResponseMetadata']['HTTPStatusCode']
+        if responseCode >= 400:
+            text_output = "Unexpected error: %s \n" % str(result)
+        else:
+            text_output = "SSL policy add to bucket: %s" % bucket_name
+
+    except ClientError as e:
+        text_output = "Unexpected error: %s \n" % e
+
+    return text_output
+
+
+# This Function find witch actions in the bucket policy and return a list with the actions that relevant to the ssl
+def get_correct_policy(policy_bucket):
+    run = policy_bucket['Statement']
+    res = []
+    options = ["s3:GetObject", "s3:*", "s3:PutObject", "s3:Put*", "s3:Get*", "*"]
+
+    for action in run:
+        if action["Effect"] == "Allow":
+            res = [ele for ele in action['Action'] if (ele in options)]
```

Comment:
avoid using 'res' and 'ele' and find long and informative names. Remember it's and open source project so your code should be very understandable

Reply:
changed the name of the variables